File: frontend/news.js
#!/bin/node

url = 'https://newsapi.org/v2/everything?apikey=a73a26f264ce491d9d5769d880731b2e'


const apiKey = 'a73a26f264ce491d9d5769d880731b2e';
const query = 'job and career'; 

// Make API request using your preferred method (e.g., fetch, axios, etc.)
fetch(`https://api.cognitive.microsoft.com/bing/v7.0/news/search?q=${query}`, {
  headers: {
    'Ocp-Apim-Subscription-Key': apiKey,
  },
})
  .then(response => response.json())
  .then(data => {
    // Accessing the first news article's URL
    const firstArticleUrl = data.value[0].url;

    // Now you can fetch the content of the news article using the obtained URL
    fetch(firstArticleUrl)
      .then(response => response.text())
      .then(articleContent => {
        // 'articleContent' now contains the HTML or text content of the news article
        console.log(articleContent);
      })
      .catch(error => {
        console.error('Error fetching article content:', error);
      });
  })
  .catch(error => {
    console.error('Error fetching news:', error);
  });

File: models/news.py
#!/usr/bin/python3
import requests


from datetime import datetime, timedelta
current_date = datetime.now()
yesterday = (current_date - timedelta(days=1)).strftime('%Y-%m-%d')
last_week = (current_date - timedelta(days=7)).strftime('%Y-%m-%d')
print("Yesterday's date:", yesterday)


url = "https://newsapi.org/v2/everything"

data = {'q': 'job AND career AND employment',
        'from': yesterday,
        }

headers = {'x-api-key': 'a73a26f264ce491d9d5769d880731b2e',
        'Content-Type': 'application/json'
        }

response = requests.get(url, params=data, headers=headers)
data = response.json()
articles = (data.get('articles'))

#url = articles[0].get('url')
#html = requests.get(url)

#print(response.json())
#rint(response.json())
#for article in articles:
#    url = (article.get('url'))
#    html = requests.get(url)
#   print(html.__dict__)
